docs/developer/architecture/control-plane-ingress.rst

```diff
@@ -0,0 +1,164 @@
+Control Plane Ingress
+=====================
+
+Context
+-------
+
+Initially, Control Plane Ingress Controller was deployed using a DaemonSet with
+a ClusterIP service using Bootstrap Control Plane IP as External IP and then
+configuring all Control Plane components using this “External IP” (like OIDC
+and various UIs).
+
+So it means, we can only reach those components using Bootstrap Control
+Plane IP, and if, for whatever reason, Bootstrap is down you can no longer
+access any UIs and you need to reconfigure all components manually, change the
+IP used everywhere to another Control Plane node IP, in order to restore
+access to Control Plane components.
+
+Here, we want to solve this issue and make Control Plane components Highly
+Available, so if you lose one node, including Bootstrap Node, you can still
+access various UIs.
+NOTE: In this document, we do not talk about the High Availability of the
+components itself but really only to the access through the Ingress
+(e.g.: We do not want to solve salt-master HA here).
+
+User Stories
+------------
+
+MetalK8s and Grafana UIs HA
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+I have a multi-node MetalK8s cluster, with at least 3 Control Plane nodes
+if I lose one of the Control Plane nodes (including the Bootstrap one)
+I can still access and authenticate on the MetalK8s and Grafana UIs.
+
+Design Choices
+--------------
+
+To have a proper HA Control Plane Ingress we want to use a
+`Virtual IP <https://en.wikipedia.org/wiki/Virtual_IP_address>`_ using
+`MetalLB <https://metallb.universe.tf/>`_ so that we can rely on layer2
+ARP requests when possible or let MetalLB advertise IPs using BGP when
+the user router support it.
```

Comment:
Do we plan to support BGP-based address management?

Reply:
I don't know, it's more a question for @thomasdanan 
But to me, since it's built-in metallb it's only some field to fill in the metallb config it's easy to "support" it

Reply:
Yes I agree, I can remove it for the moment and stay on Layer2 only. 
Maybe @xaf-scality you have an opinion on this one as well ?